test/hotspot/jtreg/vmTestbase/nsk/jvmti/scenarios/contention/TC03/tc03t001/tc03t001.cpp

```diff
@@ -127,16 +131,24 @@ static int findDeadlockThreads(jvmtiEnv* jvmti, JNIEnv* jni) {
 
         NSK_DISPLAY3("    thread #%d (%s): %p\n", i, info.name, threads[i]);
 
-        threadList[i].thread = threads[i];
-        threadList[i].dfn = -1;
-        threadList[i].name = info.name;
+        if (!strncmp(info.name, THREAD_NAME_PREFIX, THREAD_NAME_PREFIX_LEN)) {
+            NSK_DISPLAY1("Skipping thread %s\n", info.name);
+            if (!NSK_JVMTI_VERIFY(jvmti->Deallocate((unsigned char*)info.name)))
+                return NSK_FALSE;
+            continue;
+        }
+
+        threadList[debuggee_thread_cnt].thread = threads[i];
+        threadList[debuggee_thread_cnt].dfn = -1;
+        threadList[debuggee_thread_cnt].name = info.name;
+        debuggee_thread_cnt++;
     }
 
     /* deallocate thread list */
     if (!NSK_JVMTI_VERIFY(jvmti->Deallocate((unsigned char*)threads)))
         return NSK_FALSE;
 
-    for (i = 0; i < threads_count; i++) {
+    for (i = 0; i < debuggee_thread_cnt; i++) {
```

Comment:
The fix looks incomplete.
There is global `threads_count` variable and it's used in a number of places in the file.
I think it would be simpler to add local `total_thread_count` and use it only to initialize `threadList` (`GetAllThreads`, allocate `threadList`, iteration through `threads`); `debuggee_thread_cnt` needs to be replaced with `threads_count`